File: src/com/huhx0015/datastructures/Node.java
package com.huhx0015.datastructures;

/**
 * Created by Michael Yoon Huh on 4/7/2017.
 */
public class Node {

    public String name;
    public int value;

    public Node left;
    public Node right;

    public Node[] children;
    public Node[] adjacent;

    public boolean visited = false;
    public boolean marked = false;

    public Node(String name) {
        this.name = name;
    }

    public Node(int value) {
        this.value = value;
    }

//    public Node(String name, Node[] children) {
//        this.name = name;
//        this.children = children;
//    }

    public Node(String name, Node[] adjacent) {
        this.name = name;
        this.adjacent = adjacent;
    }
}

File: src/main/com/huhx0015/algorithmtester/datastructures/Node.kt
package main.com.huhx0015.algorithmtester.datastructures

/**
 * Created by Michael Yoon Huh on 4/7/2017.
 */
class Node {

    var name: String = ""
    var value: Int = 0

    var left: Node? = null
    var right: Node? = null

    var children: Array<Node>? = null
    var adjacent: Array<Node>? = null

    var visited = false
    var marked = false

    constructor(name: String) {
        this.name = name
    }

    constructor(value: Int) {
        this.value = value
    }

    constructor(name: String, adjacent: Array<Node>) {
        this.name = name
        this.adjacent = adjacent
    }
}
